src/libmongoc/tests/test-mongoc-sample-commands.c

```diff
@@ -3592,12 +3607,13 @@ _test_sample_versioned_api_example_1 (void)
     * client = mongoc_client_new (uri_sharded);
     */
 
-   client = get_client ();
+   client = get_client_for_version_api_example ();
 
    mongoc_server_api_version_from_string ("1", &server_api_version);
    server_api = mongoc_server_api_new (server_api_version);
 
-   mongoc_client_set_server_api (client, server_api, &error);
+   ok = mongoc_client_set_server_api (client, server_api, &error);
+   BSON_ASSERT (ok);
```

Comment:
Do we care much that `BSON_ASSERT` is appearing in the API examples (as opposed to the regular C `assert()`)?

Reply:
Good question. Though `BSON_ASSERT` is part of the public libbson API, and is defined in bson-macros.h, I think it makes sense to use `assert` since readers are likely to be familiar with `assert`.

I looked through other examples code for reference. There aren't many assertions, but there are instances of both uses of `BSON_ASSERT` and `assert`.

https://github.com/mongodb/mongo-c-driver/blob/3a41062e397638399a6e491ec94bd6c861ec212c/src/libmongoc/examples/example-update.c#L52:L52 uses `BSON_ASSERT`.

https://github.com/mongodb/mongo-c-driver/blob/3a41062e397638399a6e491ec94bd6c861ec212c/src/libmongoc/examples/example-gridfs.c#L56:L56 uses `assert`.